src/tokenizer/index.js

```diff
@@ -138,8 +153,8 @@ export default class Tokenizer {
   readToken(code) {
     // Identifier or keyword. '\uXXXX' sequences are allowed in
     // identifiers, so '\' also dispatches to that.
-    if (isIdentifierStart(code) || code === 92 /* '\' */) {
-      return this.readWord();
+    if (isIdentifierStart(code) || code === 92) {
+      /* '\' */ return this.readWord();
```

Comment:
This is not intended, I just created an issue for it: https://github.com/prettier/prettier/issues/867